src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -891,6 +892,18 @@ private function registerFormConfiguration(array $config, ContainerBuilder $cont
             $container->setParameter('form.type_extension.csrf.enabled', false);
         }
 
+        if ($config['form']['use_attribute']) {
+            $loader->load('form_metadata.php');
+
+            $container->registerAttributeForAutoconfiguration(AsFormType::class, static function (ChildDefinition $definition, AsFormType $attribute, \ReflectionClass $ref) {
+                $definition
+                    ->addTag('container.excluded.form.metadata.form_type', ['class_name' => $ref->getName()])
+                    ->addTag('container.excluded')
+                    ->setAbstract(true)
+                ;
+            });
+        }
```

Comment:
The new `addResourceTag()` method should be used; the related `findTaggedResourceIds` is already used in the compiler pass.

As of now, have not tag name prefixed with `container.excluded.`. I don't think that's necessary, since there's also the `container.excluded` tag.

I think reading the class name from the service definition in the compiler pass is more resilient. It is also easier to tag a service "manually" without repeating the class name in the tag if someone don't use autoconfiguration.


```suggestion
            $container->registerAttributeForAutoconfiguration(AsFormType::class, static function (ChildDefinition $definition) {
                $definition->addResourceTag('form.metadata.form_type');
            });
        }
```

Reply:
IMO it better to keep it as is, not using `addResourceTag` because it breaks low-deps tests. WDYT?